bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/discovery/AtlonaDiscoveryParticipant.java

```diff
@@ -0,0 +1,128 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.atlona.internal.pro3.AtlonaPro3Config;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.sddp.SddpDevice;
+import org.openhab.core.config.discovery.sddp.SddpDiscoveryParticipant;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Discovery Service for Atlona HDMI matrices that support SDDP.
+ *
+ * @author Michael Lobstein - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class AtlonaDiscoveryParticipant implements SddpDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(AtlonaDiscoveryParticipant.class);
+
+    private static final String ATLONA = "Atlona";
+    private static final String PROXY_AVSWITCH = "avswitch";
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Set.of(THING_TYPE_PRO3_44M, THING_TYPE_PRO3_66M, THING_TYPE_PRO3_88M, THING_TYPE_PRO3_1616M,
+                THING_TYPE_PRO3HD_44M, THING_TYPE_PRO3HD_66M);
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(SddpDevice device) {
+        final ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            final Map<String, Object> properties = new HashMap<>(2);
+            final String label = device.model + " (" + device.ipAddress + ")";
+
+            properties.put(Thing.PROPERTY_MAC_ADDRESS, uid.getId());
+            properties.put(AtlonaPro3Config.IP_ADDRESS, device.ipAddress);
+
+            final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withRepresentationProperty(Thing.PROPERTY_MAC_ADDRESS).withLabel(label).build();
+
+            logger.debug("Created a DiscoveryResult for device '{}' with UID '{}'", label, uid.getId());
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    /*
+     * The Atlona SDDP message has the following format
+     *
+     * <pre>
+     * NOTIFY ALIVE SDDP/1.0
+     * From: "192.168.1.30:1902"
+     * Host: "AT-UHD-PRO3-88M_B898B0030F4D"
+     * Type: "AT-UHD-PRO3-88M"
+     * Max-Age: 1800
+     * Primary-Proxy: "avswitch"
+     * Proxies: "avswitch"
+     * Manufacturer: "Atlona"
+     * Model: "AT-UHD-PRO3-88M"
+     * Driver: "avswitch_Atlona_AT-UHD-PRO3-88M_IP.c4i"
+     * Config-URL: "http://192.168.1.30/"
+     * </pre>
+     */
+    @Override
+    public @Nullable ThingUID getThingUID(SddpDevice device) {
+        if (ATLONA.equals(device.manufacturer) && PROXY_AVSWITCH.equals(device.primaryProxy)
+                && !device.macAddress.isBlank()) {
```

Comment:
You could also check that device.ipAddress is not blank.

Reply:
done